modules/core-components/src/main/java/org/mule/runtime/core/internal/routing/ParallelForEach.java

```diff
@@ -127,4 +125,15 @@ protected ForkJoinStrategyFactory getDefaultForkJoinStrategyFactory() {
   public void setCollectionExpression(String collectionExpression) {
     this.collectionExpression = collectionExpression;
   }
+
+  private Message createMessage(TypedValue<?> partTypedValue, CoreEvent event) {
+    if (featureFlaggingService.isEnabled(PARALLEL_FOREACH_FLATTEN_MESSAGE) && partTypedValue.getValue() instanceof Message) {
```

Comment:
Added the feature flag here again instead of joining the case with the one on manageTypedValuePayload to consider cases in which the flag is on but the partTypedValue is not a Message